app/src/main/java/cm/aptoide/pt/home/apps/AppsFragment.java

```diff
@@ -100,13 +102,11 @@ public static AppsFragment newInstance() {
     buildIgnoreUpdatesDialog();
     userAvatar = (ImageView) view.findViewById(R.id.user_actionbar_icon);
 
-    attachPresenter(new AppsPresenter(this, new AppsManager(new UpdatesManager(
-        RepositoryFactory.getUpdateRepository(getContext(),
-            ((AptoideApplication) getContext().getApplicationContext()).getDefaultSharedPreferences())),
-        ((AptoideApplication) getContext().getApplicationContext()).getInstallManager(),
-        new AppMapper(), downloadAnalytics, installAnalytics, updatesAnalytics,
-        getContext().getPackageManager(), getContext(), new DownloadFactory(
-        ((AptoideApplication) getContext().getApplicationContext()).getMarketName())),
+    attachPresenter(new AppsPresenter(this,
+        new AppsManager(updatesManager, installManager, new AppMapper(), downloadAnalytics,
+            installAnalytics, updatesAnalytics, getContext().getPackageManager(), getContext(),
+            new DownloadFactory(
+                ((AptoideApplication) getContext().getApplicationContext()).getMarketName())),
```

Comment:
Need to be careful with this changes because this is not in the scope of the appview and needs to be tested by ruben after.